skimage/measure/sopoly.py

```diff
@@ -0,0 +1,1776 @@
+    min_crest_ids : list
+        A list of indices of the minimum crest vertices.
+    max_crest : int
+        The positional index of the maximum crest vertex.
+    min_crest : int
+        The positional index of the minimum crest vertex.
+    """
+    # Get the direction of the polygon when traversing its perimeter.
+    edges = vertices[1:, :2] - vertices[:-1, :2]
+    edges = np.vstack((vertices[0, :2] - vertices[-1, :2], edges))
+    edges = edges / np.linalg.norm(edges, axis=1).reshape(-1, 1)
+
+    norm_left = np.hstack((-edges[:, np.newaxis, 1], edges[:, np.newaxis, 0]))
+
+    cos_angle = np.sum(norm_left[:-1] * edges[1:], axis=1)
+    cos_angle = np.append(cos_angle, np.sum(norm_left[-1] * edges[0]))
+
+    dir_left = cos_angle > 0.0
+
+    # Only crest points of left turns are considered
+    diff_y_prev = vertices[1:, 1] - vertices[:-1, 1]
+    diff_y_next = -diff_y_prev
+    diff_y_prev = np.insert(diff_y_prev, 0, vertices[0, 1] - vertices[-1, 1])
+    diff_y_next = np.append(diff_y_next, vertices[-1, 1] - vertices[0, 1])
+
+    u_prev = diff_y_prev > tolerance
+    u_next = diff_y_next > tolerance
+    d_prev = diff_y_prev < -tolerance
+    d_next = diff_y_next < -tolerance
+
+    ey_next = np.fabs(diff_y_next) <= tolerance
+
+    # Determines if the path is climbing up.
+    clmb_up = reduce(lambda l1, l2:
+                     l1 + [(l2[0] and l2[2])
+                           or (l1[-1] and not l2[0] and not l2[1])],
+                     zip(u_prev, d_prev, ey_next), [False])[1:]
+
+    # Determines if the path is climbing down.
+    clmb_dwn = reduce(lambda l1, l2:
+                      l1 + [(l2[0] and l2[2])
+                            or (l1[-1] and not any(l2[:2]))],
+                      zip(d_prev, u_prev, ey_next), [False])[1:]
+
+    # Find maximum crests on left turns only.
+    max_crest_ids = np.nonzero(dir_left
+                               * (u_prev * u_next + u_prev * clmb_up))[0]
+
+    # Find minimum crests on left turns only.
+    min_crest_ids = np.nonzero(dir_left
+                               * (d_prev * d_next + d_prev * clmb_dwn))[0]
+
+    if len(max_crest_ids) > 0:
+        max_crest = max_crest_ids[np.argmax(vertices[max_crest_ids, 1])]
+    else:
+        max_crest = None
+
+    if len(min_crest_ids) > 0:
+        min_crest = min_crest_ids[np.argmin(vertices[min_crest_ids, 1])]
+    else:
+        min_crest = None
+
+    return max_crest_ids, min_crest_ids, max_crest, min_crest
+
+
+def _check_clockwise(vertices):
+    """Check whether the polygon vertices are in a clockwise ordering.
+    The polygon sub division algorithm works for clockwise polygns.
+
+    Parameters
+    ----------
+    vertices : numpy.ndarray
+        An array containing the coordinates of the polygon vertices.
+
+    Returns
+    -------
+    is_clockwise : bool or None
+        Whether the polygon is in clockwise direction or not.
+        This is None when no crest points were detected.
```

Comment:
With the current implementation I don't see a way for this function to return `None`. Am I missing something?